lib/packages/core/lib/src/views/participants/participants.dart

```diff
@@ -0,0 +1,288 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:infinite_scroll_pagination/infinite_scroll_pagination.dart';
+import 'package:likeminds_chat_flutter_core/likeminds_chat_flutter_core.dart';
+import 'package:likeminds_chat_flutter_core/src/blocs/participants/participants_bloc.dart';
+import 'package:likeminds_chat_flutter_ui/likeminds_chat_flutter_ui.dart';
+
+/// [LMChatroomParticipantsPage] is a page to display the participants of a chat room.
+/// It uses [LMChatParticipantsBloc] to fetch the participants of a chat room.
+class LMChatroomParticipantsPage extends StatefulWidget {
+  /// [chatroomViewData] is the chat room for which the participants are to be fetched.
+  final LMChatRoomViewData chatroomViewData;
+
+  /// [firstPageErrorIndicatorBuilder] is a builder to show the error in fetching the first page of participants.
+  final Widget Function(BuildContext)? firstPageErrorIndicatorBuilder;
+
+  /// [newPageErrorIndicatorBuilder] is a builder to show the error in fetching the new page of participants.
+  final Widget Function(BuildContext)? newPageErrorIndicatorBuilder;
+
+  /// [firstPageProgressIndicatorBuilder] is a builder to show the progress indicator while fetching the first page of participants.
+  final Widget Function(BuildContext)? firstPageProgressIndicatorBuilder;
+
+  /// [newPageProgressIndicatorBuilder] is a builder to show the progress indicator while fetching the new page of participants.
+  final Widget Function(BuildContext)? newPageProgressIndicatorBuilder;
+
+  /// [noItemsFoundIndicatorBuilder] is a builder to show the indicator when no participants are found.
+  final Widget Function(BuildContext)? noItemsFoundIndicatorBuilder;
+
+  /// [noMoreItemsIndicatorBuilder] is a builder to show the indicator when no more participants are found.
+  final Widget Function(BuildContext)? noMoreItemsIndicatorBuilder;
+
+  /// [appBarBuilder] is a builder to build the app bar for the page.
+  final LMChatAppBarBuilder? appBarBuilder;
+
+  /// [userTileBuilder] is a builder to build the user tile for the participants.
+  final Widget Function(BuildContext, LMChatUserTile)? userTileBuilder;
```

Comment:
We should use a typedef here as well. But of type `LMChatTile`.

Reply:
done